packages/hawtio/src/plugins/camel/routes-service.ts

```diff
@@ -205,6 +256,102 @@ class RoutesService {
+      deltaProcessingTime: parseInt(pDoc.getAttribute('deltaProcessingTime') ?? '0'),
+      exchangeInflight: parseInt(pDoc.getAttribute('exchangeInflight') ?? '0'),
+      externalRedeliveries: parseInt(pDoc.getAttribute('externalRedeliveries') ?? '0'),
+      firstExchangeCompleteTimestamp: parseInt(pDoc.getAttribute('firstExchangeCompleteTimestamp') ?? '0'),
+      firstExchangeCompletedExchangeId: parseInt(pDoc.getAttribute('firstExchangeCompletedExchangeId') ?? '0'),
+      state: pDoc.getAttribute('state'),
+      index: parseInt(pDoc.getAttribute('index') ?? '0'),
+      lastExchangeCompleteTimestamp: parseInt(pDoc.getAttribute('lastExchangeCompleteTimestamp') ?? '0'),
+      maxProcessingTime: parseInt(pDoc.getAttribute('maxProcessingTime') ?? '0'),
+      meanProcessingTime: parseInt(pDoc.getAttribute('meanProcessingTime') ?? '0'),
+      minProcessingTime: parseInt(pDoc.getAttribute('minProcessingTime') ?? '0'),
+      redeliveries: parseInt(pDoc.getAttribute('redeliveries') ?? '0'),
+      resetTimestamp: parseInt(pDoc.getAttribute('resetTimestamp') ?? '0'),
+      sourceLineNumber: parseInt(pDoc.getAttribute('sourceLineNumber') ?? '0'),
+      startTimestamp: parseInt(pDoc.getAttribute('startTimestamp') ?? '0'),
+    }
+  }
+
+  createRouteStats(pDoc: Element): RouteStats {
+    const procStats = pDoc.getElementsByTagName('processorStat')
+    const processorStats: ProcessorStats[] = []
+    for (const pStat of procStats) {
+      processorStats.push(this.createProcessorStats(pDoc.id, pStat))
+    }
+
+    return {
+      id: pDoc.id,
+      deltaProcessingTime: parseInt(pDoc.getAttribute('deltaProcessingTime') ?? '0'),
+      exchangesCompleted: parseInt(pDoc.getAttribute('exchangesCompleted') ?? '0'),
+      exchangesFailed: parseInt(pDoc.getAttribute('exchangesFailed') ?? '0'),
+      exchangesInflight: parseInt(pDoc.getAttribute('exchangesInflight') ?? '0'),
+      externalRedeliveries: parseInt(pDoc.getAttribute('externalRedeliveries') ?? '0'),
+      failuresHandled: parseInt(pDoc.getAttribute('failuresHandled') ?? '0'),
+
+      lastProcessingTime: parseInt(pDoc.getAttribute('lastProcessingTime') ?? '0'),
+      maxProcessingTime: parseInt(pDoc.getAttribute('maxProcessingTime') ?? '0'),
+      meanProcessingTime: parseInt(pDoc.getAttribute('meanProcessingTime') ?? '0'),
+      minProcessingTime: parseInt(pDoc.getAttribute('minProcessingTime') ?? '0'),
+      processorStats: processorStats,
+      redeliveries: parseInt(pDoc.getAttribute('redeliveries') ?? '0'),
+      resetTimestamp: parseInt(pDoc.getAttribute('resetTimestamp') ?? '0'),
+      sourceLocation: pDoc.getAttribute('sourceLocation'),
+      startTimestamp: parseInt(pDoc.getAttribute('startTimestamp') ?? '0'),
+      state: pDoc.getAttribute('state'),
+      totalProcessingTime: parseInt(pDoc.getAttribute('totalProcessingTime') ?? '0'),
+      firstExchangeCompletedExchangeId: pDoc.getAttribute('firstExchangeCompletedExchangeId'),
```

Comment:
Likewise, it seems that maybe a loop over `pDoc` attributes would be less verbose?

Reply:
A more cleaner way to avoid a lot of boilerplate, taking into account refactoring the parseInt part to its own function.

```
Const obj = Object.fromEntries([attributeName,
attributeName2,
...].map(attribute => getFromPdoc(attribute))

Obj.id =pdoc.id

Return obj
```

If there are any special case can be added to the object like with the id.


Anothwr way

```
Const attributesToRetrieve =[attr1, attr2, attr3...]

Return {
id: p.id
...Object.fromEntries(attributesToRetrieve.map(getFromPdoc))
```